From bcc125d1bdd969897bc7719913afefc64931efe4 Mon Sep 17 00:00:00 2001
From: fossfreedom <foss.freedom@gmail.com>
Date: Sun, 17 Sep 2017 11:25:05 +0100
Subject: [PATCH] Style and icon size fixes #3

---
 src/applet-main.c | 20 ++++++++++++++++++--
 src/applet.c      | 32 ++++++++++++++++++++++++++++----
 2 files changed, 46 insertions(+), 6 deletions(-)

diff --git a/src/applet-main.c b/src/applet-main.c
index f6e0c21..4839da9 100644
--- a/src/applet-main.c
+++ b/src/applet-main.c
@@ -216,6 +216,7 @@ static void entry_added(IndicatorObject *io, IndicatorObjectEntry *entry, GtkWid
         gboolean something_visible = FALSE;
         gboolean something_sensitive = FALSE;
         GtkStyleContext *context;
+        GtkCssProvider *css_provider = NULL;
         GSettings *settings = NULL;
 
         /*
@@ -249,6 +250,7 @@ static void entry_added(IndicatorObject *io, IndicatorObjectEntry *entry, GtkWid
 
         if (entry->image != NULL) {
                 g_debug("zzz have an image");
+                gtk_image_set_pixel_size(entry->image, 22);
                 gtk_box_pack_start(GTK_BOX(box), GTK_WIDGET(entry->image), FALSE, FALSE, 1);
                 if (gtk_widget_get_visible(GTK_WIDGET(entry->image))) {
                         g_debug("zzz and is visible");
@@ -308,8 +310,22 @@ static void entry_added(IndicatorObject *io, IndicatorObjectEntry *entry, GtkWid
         */
         context = gtk_widget_get_style_context(GTK_WIDGET(menuitem));
         gtk_style_context_add_class(context, "budgie-menubar");
-        //context = gtk_widget_get_style_context(GTK_WIDGET(menubar));
-        //gtk_style_context_remove_class(context, "menubar");
+        context = gtk_widget_get_style_context(GTK_WIDGET(menubar));
+        gtk_style_context_remove_class(context, "menubar");
+
+        css_provider = gtk_css_provider_new();
+        gtk_css_provider_load_from_data(css_provider,
+                                        ".budgie-menubar { \n"
+                                        "    padding-left: 2px; \n"
+                                        "    padding-right: 2px; \n"
+                                        "} \n",
+                                        -1,
+                                        NULL);
+        gtk_style_context_add_provider(GTK_STYLE_CONTEXT(gtk_widget_get_style_context(
+                                           GTK_WIDGET(menuitem))),
+                                       GTK_STYLE_PROVIDER(css_provider),
+                                       GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+
         g_debug("zzz adding budgie-menubar");
 
         gtk_container_add(GTK_CONTAINER(menuitem), box);
diff --git a/src/applet.c b/src/applet.c
index 9118098..3ca581a 100644
--- a/src/applet.c
+++ b/src/applet.c
@@ -69,15 +69,39 @@ static void appindicator_applet_init(AppIndicatorApplet *self)
 {
         GtkWidget *eventbox = NULL;
         GtkWidget *menubar = NULL;
-        GtkStyleContext *context;
+        GtkCssProvider *css_provider = NULL;
 
         gint indicators_loaded = 0;
 
         menubar = gtk_menu_bar_new();
+        css_provider = gtk_css_provider_new();
+#if GTK_CHECK_VERSION(3, 20, 0)
+        gtk_css_provider_load_from_data(css_provider,
+                                        "menubar { \n"
+                                        "    background: transparent; } \n"
+                                        ".budgie-menubar { \n"
+                                        "    padding-left: 2px; \n"
+                                        "    padding-right: 2px; \n"
+                                        "} \n",
+                                        -1,
+                                        NULL);
+#else
+        gtk_css_provider_load_from_data(css_provider,
+                                        ".menuitem { \n"
+                                        "    background: transparent; } \n"
+                                        ".budgie-menubar { \n"
+                                        "    padding-left: 2px; \n"
+                                        "    padding-right: 2px; \n"
+                                        "} \n",
+                                        -1,
+                                        NULL);
+#endif
+        gtk_style_context_add_provider(GTK_STYLE_CONTEXT(gtk_widget_get_style_context(
+                                           GTK_WIDGET(menubar))),
+                                       GTK_STYLE_PROVIDER(css_provider),
+                                       GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+
         eventbox = gtk_event_box_new();
-        context = gtk_widget_get_style_context(GTK_WIDGET(menubar));
-        gtk_style_context_add_class(context, "budgie-panel");
-        
         gtk_container_add(GTK_CONTAINER(self), eventbox);
         gtk_widget_show(eventbox);
 
-- 
2.7.4

